lemminx-liberty/src/main/java/io/openliberty/lemminx/liberty/LibertyCompletionParticipant.java

```diff
@@ -69,16 +71,22 @@ private CompletionItem buildFeatureCompletionItem(Feature feature, DOMElement fe
         return item;
     }
 
-    private List<CompletionItem> buildCompletionItems(DOMElement featureElement, DOMDocument document,
+    private List<CompletionItem> buildCompletionItems(DOMElement featureElement, DOMDocument domDocument,
             List<String> existingFeatures) {
-        final String libertyVersion = SettingsService.getInstance().getLibertyVersion();
+
+        String libertyVersion = LibertyUtils.getVersion(domDocument);
+        if (libertyVersion == null) {
+            // try to get version from installed Liberty
+            libertyVersion = LibertyUtils.getVersion(domDocument);
+        }
```

Comment:
Seems to be calling `LibertyUtils.getVersion(domDocument);` twice? Is this on purpose? 

other files seem to do this instead
```
        String libertyVersion = SettingsService.getInstance().getLibertyVersion();
        if (libertyVersion == null) {
            // try to get version from installed Liberty
            libertyVersion = LibertyUtils.getVersion(domDocument);
        }
```